### ISSUE TITLE
How batch size effect the model?

### ISSUE THREAD

**Issue by 1292765944** [weiliu89/caffe#175]
Dear Wei Liu:
    In cnn, the loss backward are calculated by a batch of input, so what the effect of large batch size, and could it be replaced by the iter_size parameter in solver.prototxt completely? Thank you!

**Comment by weiliu89** [comment]
It depends on your normalization method (e.g. there is BATCH, VALID, etc.) Different normalization has different effect when you use the accumulate gradient trick provided by iter_size parameter. The iter_size does nothing but compute the mean gradient within given iter_size.

**Comment by 1292765944** [comment]
@weiliu89 oh, thanks! But I wonder what's the special effect of the iter_size on batch normalisation?

**Comment by nopattern** [comment]
someone has done special research on batch_size on mnist. Google it by yourself. Too small  and too large has bad effect either.

**Comment by 1292765944** [comment]
@nopattern Oh, Thank you, But my question is whether the iter_size can replace batch_size parameter?

**Comment by liu09114** [comment]
Yes, iter_size*batch_size is the real batch_size. So if you only have small memory, you can make small batch_size and large iter_size.
